scripts/rescript_format.js

```diff
@@ -164,20 +214,34 @@ function main(argv, rescript_exe, bsc_exe) {
         }
       }
       var hasError = false;
+      var incorrectlyFormattedFiles = 0;
       files.forEach(file => {
-        var write = isSupportedFile(file);
+        var write = isSupportedFile(file) && !check.val;
         var flags = write ? ["-o", file, "-format", file] : ["-format", file];
-        child_process.execFile(bsc_exe, flags, (error, stdout, stderr) => {
-          if (error === null) {
-            if (!write) {
-              process.stdout.write(stdout);
+        try {
+          const formatted = child_process.execFileSync(bsc_exe, flags);
```

Comment:
As discussed on Discord, please add a comment explaining why processing is sync here, but async above.

Reply:
Side comment: do we have a discord community channel? 😲

Reply:
@mattdamon108 haha no dm

Reply:
@tsnobip Ah! haha

Reply:
@mattdamon108 What's your Discord handle?

Reply:
@cknitt moondaddi#2730

Reply:
> As discussed on Discord, please add a comment explaining why processing is sync here, but async above.

now that I think about it, I don't know if the asynchronous processing we had before actually created any issue? I'll indeed add some comments about my reasoning.

Reply:
I rewrote it to use the same logic, there's no reason why the output would be mixed anyway.